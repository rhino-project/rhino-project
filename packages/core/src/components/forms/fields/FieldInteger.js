import FieldInput from './FieldInput';

// We want it to be blank if it's not a number
const setValueAs = (value) => {
  const parsed = parseInt(value, 10);
  if (isNaN(parsed)) return null;

  return parsed;
};

export const FieldInteger = (props) => (
  <FieldInput type="number" setValueAs={setValueAs} {...props} />
);

export default FieldInteger;
