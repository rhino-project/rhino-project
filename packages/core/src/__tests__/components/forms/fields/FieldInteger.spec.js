import FieldInteger from 'rhino/components/forms/fields/FieldFloat';
import { sharedFieldTests } from './sharedFieldTests';

describe('FieldInteger', () => {
  sharedFieldTests(FieldInteger);
});
