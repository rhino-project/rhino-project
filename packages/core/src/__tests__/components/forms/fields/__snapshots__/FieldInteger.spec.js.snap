// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FieldInteger > render with disabled 1`] = `
<DocumentFragment>
  <input
    aria-invalid="false"
    autocomplete="off"
    class="form-control"
    disabled=""
    id="dummy"
    name="dummy"
    path="dummy"
    type="number"
  />
</DocumentFragment>
`;

exports[`FieldInteger > render with place holder 1`] = `
<DocumentFragment>
  <input
    aria-invalid="false"
    autocomplete="off"
    class="form-control"
    id="dummy"
    name="dummy"
    path="dummy"
    placeholder="placeholder"
    type="number"
  />
</DocumentFragment>
`;

exports[`FieldInteger > renders inside of model context 1`] = `
<DocumentFragment>
  <input
    aria-invalid="false"
    autocomplete="off"
    class="form-control"
    id="dummy"
    name="dummy"
    path="dummy"
    type="number"
  />
</DocumentFragment>
`;

exports[`FieldInteger > should render with global override shorthand 1`] = `
<DocumentFragment>
  <div>
    Bar
  </div>
</DocumentFragment>
`;
